### ISSUE TITLE
Does not detect duplicate checksum

### ISSUE THREAD

**Issue by stain** [ini-bdds/minid#40]
The "Checking if the entity.. exists on the server" routine does not seem to work:

```
stain@biggie:~/Downloads$ minid --register --title "CWL run of Somatic Variant Calling Workflow (CWLProv 0.5.0 Research Object)" somaticwf_0.5.0_mac.zip 
2019-01-24 13:38:39,540 - INFO - Computing checksum for somaticwf_0.5.0_mac.zip using <sha256 HASH object @ 0x7fadadcede90>
2019-01-24 13:38:40,022 - INFO - Checking if the entity f76c272c932c4d0a0b4536b7e595222b2e9d9f16465203e146004c39cb974042 already exists on the server: http://minid.bd2k.org/minid
2019-01-24 13:38:40,292 - INFO - Creating new identifier
2019-01-24 13:38:42,039 - INFO - Created/updated minid: minid:b9t14c
```

But second run with same file, while it finds the same hash `f76c272c932c4d0a0b4536b7e595222b2e9d9f16465203e146004c39cb974042` it still goes no _Creating new identifier_, this time `minid:b9p712` instead of previous `minid:b9t14c`

```
stain@biggie:~/Downloads$ minid --register --title "CWL run of Somatic Variant Calling Workflow (CWLProv 0.5.0 Research Object)" somaticwf_0.5.0_mac.zip 
2019-01-24 13:38:49,792 - INFO - Computing checksum for somaticwf_0.5.0_mac.zip using <sha256 HASH object @ 0x7f40bebbee90>
2019-01-24 13:38:49,905 - INFO - Checking if the entity f76c272c932c4d0a0b4536b7e595222b2e9d9f16465203e146004c39cb974042 already exists on the server: http://minid.bd2k.org/minid
2019-01-24 13:38:50,125 - INFO - Creating new identifier
2019-01-24 13:38:51,732 - INFO - Created/updated minid: minid:b9p712
```

## Install info

* Ubuntu 16.04
* Conda

```
stain@biggie:~/Downloads$ pip show minid
Name: minid
Version: 1.3.0
Summary: BD2K Minimum Viable Identifier
Home-page: http://minid.bd2k.org/
Author: Kyle Chard
Author-email: chard@uchicago.edu
License: Apache 2.0
Location: /home/stain/software/miniconda/lib/python3.6/site-packages
Requires: requests
Required-by: 
stain@biggie:~/Downloads$ python -V
Python 3.6.4 :: Anaconda, Inc.
```

**Comment by kylechard** [comment]
The behavior for creating new minids was changed some time back. Originally, we had the model that we would only create one minid for a checksum, any new attempts to create a minid would be associated with the original one. We decided for several reasons (e.g., private minids, adding messy/inaccurate metadata, etc.) that it would instead be better to create a new minid unless the user specifically requested to modify/extend an existing minid. 

So the behavior here is by design. But the messages you are seeing look like a bug to me (in that it shouldn't be looking up other minids if not needed). 

BTW, we're working on a new client which will tidy up these interactions and add some more advanced auth support. That should be out in the next week or so if you're interested in trying it.
